### ISSUE TITLE
Cannot override rules with dependsOn field in action

### ISSUE THREAD

**Issue by simpsons310** [laravel/nova-issues#4714]
- Laravel Version: 2.21.0
- Nova Version: 4.12.13
- PHP Version: 8.1.8
- Database Driver & Version:
- Operating System and Version: mysql:8.0
- Browser type and version: Google Chrome 104.0.5112.101 (Official Build) (64-bit)

### Description:
I have an action, and this is my list fields definition:
```
public function fields(NovaRequest $request)
{
    return [
         Select::make(__('Control Mode'), 'command_mode')
                ->rules('required')
                ->options($commands)
                ->displayUsingLabels(),
            Number::make(__('Timeout'), 'session_timeout')
                ->rules('integer', 'min:1', 'max:1000')
                ->default(10)
                ->hide()
                ->dependsOn(['command_mode'], function (Number $field, NovaRequest $request, FormData $formData) {
                    if ($formData->command_mode === config('access-control.entry.command.open')) {
                        $field->rules('min:10')->show();
                    }
                }),
        ];
    }
```

When I input value `1` into `Timeout` field then execute action, action still execute successfully.

This doesn't work `required` rule too, I have to use `required()` method instead. I think this should have been fixed in issue [4170](https://github.com/laravel/nova-issues/issues/4170)

**Comment by crynobone** [comment]
Sorry, I can't follow the description. Can you show a video or create a full reproducing repository to demonstrate the issue?

**Comment by simpsons310** [comment]
@crynobone

![test - Made with Clipchamp](https://user-images.githubusercontent.com/34018204/185541385-24936b3c-f4c9-453f-8c93-504ee9861055.gif)

With above code, action should throw validation exception for `Access Entry Timeout` field. `Access Entry Timeout`'s value must be greater or equal `10`. But action execute successfully with value `1`

**Comment by crynobone** [comment]
There still missing information here. Does `$formData->command_code` exactly match the configuration? 

This is why it always good to add reproducing repository when reporting a bug.

**Comment by simpsons310** [comment]
> Does $formData->command_code exactly match the configuration?

Yes, it does. `Timeout` field will show when `command_code  = 1`.

Actually, initially I wanted to test with the `required` rule, but it didn't work so I wanted to try testing with other rules. So in the code above I put the rule `min:10`.  I've edited code above, please check again.

> This is why it always good to add reproducing repository when reporting a bug.
Sorry, but i cannot create another laravel nova project because i don't have license key
